src/Workleap.OpenApi.MSBuild/ValidateOpenApiTask.cs

```diff
@@ -4,28 +4,43 @@ namespace Workleap.OpenApi.MSBuild;
 
 public sealed class ValidateOpenApiTask : CancelableAsyncTask
 {
+    private const string CodeFirst = "CodeFirst";
+    private const string ContractFirst = "ContractFirst";
+    
+    /// <summary>
+    /// 2 supported mode:
+    ///     - CodeFirst: Generate the OpenAPI specification files from the code
+    ///     - ContractFirst: Will use the OpenAPI specification files provided
+    /// </summary>
+    [Microsoft.Build.Framework.Required]
+    public string DevelopmentMode { get; set; } = string.Empty;
+
+    /// <summary>When Development mode is Contract first, will validate if the specification match the code.</summary>  
+    [Microsoft.Build.Framework.Required]
+    public bool ValidateCodeSync { get; set; } = false;
```

Comment:
Please prefix the properties with "OpenApi". It should be clear in the consumer code that a MSBuild property is specific to this tool and not a built-in MSBuild property